### ISSUE TITLE
Install latest version (v0.10.7) Mediapipe on MacOS 14.0 

### ISSUE THREAD

**Issue by BrandlMax** [google/mediapipe#4888]
### OS Platform and Distribution

MacOS 14.0 (Sonoma) - Intel

### Programming Language and version

Python 3.11.5

### Installed using virtualenv? pip? Conda?(if python)

pip

### MediaPipe version

0.9.1 -> 0.10.7

### Android AAR (if android)

None

### OpenCV version (if running on desktop)

4.8.1.78

### Describe the problem

Can't install newest version of mediapipe 

### Complete Logs

```shell
Hey 👋, I try to use PoseLandmarker on MacOS with Python.

Initial ERROR:

AttributeError: module 'mediapipe.tasks.python.vision' has no attribute 'PoseLandmarkerOptions'
```

My assumption then was that it is probably because my mediapipe version is too old.
I get the following error when trying to install the latest version of MediaPipe:

```
pip install mediapipe==0.10.7

ERROR: Could not find a version that satisfies the requirement mediapipe==0.10.7 (from versions: 0.9.1.0)

ERROR: No matching distribution found for mediapipe==0.10.7
```

When trying to install manually using the wheels file I get this error:

```
pip install /mediapipe-0.10.7-cp311-cp311-macosx_11_0_universal2.whl 

ERROR: mediapipe-0.10.7-cp311-cp311-macosx_11_0_universal2.whl is not a supported wheel on this platform.
```

I think its related to: 
https://github.com/google/mediapipe/issues/4882

Thank you!
```

**Comment by BrandlMax** [comment]
@kuaashish 
Sure, its:
```
2,9 GHz 6-Core Intel Core i9
```

**Comment by schmidt-sebastian** [comment]
Sorry - I don't have an Intel Mac nor am I able to upgrade to 14.0 yet. But this looks like something we need to fix ASAP. By any chance, can you confirm that this was working with Mac OS 13.x?

**Comment by BrandlMax** [comment]
@schmidt-sebastian Unfortunately not, sorry, just started with the project on 14.0

**Comment by schmidt-sebastian** [comment]
I /think/ that https://github.com/google/mediapipe/commit/2d0d258403faa417aa1c6d445e88db86e2c7ffb5 fixes this. You can try this manually by deleting these files from the wheel:

./mediapipe/tasks/python/metadata/flatbuffers_lib/_pywrap_flatbuffers/arm64.cpython-38-darwin.so
./mediapipe/tasks/cc/metadata/python/_pywrap_metadata_version/arm64.cpython-38-darwin.so
./mediapipe/python/_framework_bindings/arm64.cpython-38-darwin.so

**Comment by kuaashish** [comment]
Hi @BrandlMax,

To achieve this, use the Wheel Python library to unpack the .whl file. First, install Wheel with the command:

`pip install wheel
`
Next, navigate to the `.whl` file path and execute the command:

`wheel unpack /mediapipe-0.10.7-cp311-cp311-macosx_11_0_universal2.whl`

Afterwards, go to the unpacked file, located in the same folder as the .`whl` file. Delete the following three files as indicated below:

> ./mediapipe/tasks/python/metadata/flatbuffers_lib/_pywrap_flatbuffers/arm64.cpython-38-darwin.so
> ./mediapipe/tasks/cc/metadata/python/_pywrap_metadata_version/arm64.cpython-38-darwin.so
> ./mediapipe/python/_framework_bindings/arm64.cpython-38-darwin.so

Once deleted, repack the files using the command:

`wheel pack modified_file_name`

This will generate the modified `.whl `file. Finally, attempt manual installation using pip as you did previously. Additional, Please inform us if now you are able to build using pip.

Thank you

**Comment by BrandlMax** [comment]
I followed your steps, but I still got an error.

```
pip install mediapipe-0.10.7-cp311-cp311-macosx_13_0_x86_64.whl 
ERROR: mediapipe-0.10.7-cp311-cp311-macosx_13_0_x86_64.whl is not a supported device on this platform.
```

I found a hint [here](https://stackoverflow.com/questions/38866758/filename-whl-is-not-a-supported-wheel-on-this-platform) that it has something to do with naming, so I used ``pip debug --verbose`` to find out the compatible names. 

So after renaming the file to
```
mediapipe-0.10.7-cp311-cp311-macosx_10_16_x86_64.whl
```
it works! 

Thank you so much for your quick reply and solution!
